processor/src/main/java/org/mapstruct/ap/internal/model/common/ImplementationType.java

```diff
@@ -16,23 +16,29 @@ public class ImplementationType {
     private final Type type;
     private final boolean initialCapacityConstructor;
     private final boolean loadFactorAdjustment;
+    private final String factoryMethodName;
 
-    private ImplementationType(Type type, boolean initialCapacityConstructor, boolean loadFactorAdjustment) {
+    private ImplementationType(
+        Type type,
+        boolean initialCapacityConstructor,
+        boolean loadFactorAdjustment,
+        String factoryMethodName) {
         this.type = type;
         this.initialCapacityConstructor = initialCapacityConstructor;
         this.loadFactorAdjustment = loadFactorAdjustment;
+        this.factoryMethodName = factoryMethodName;
     }
 
     public static ImplementationType withDefaultConstructor(Type type) {
-        return new ImplementationType( type, false, false );
+        return new ImplementationType( type, false, false, null );
     }
 
     public static ImplementationType withInitialCapacity(Type type) {
-        return new ImplementationType( type, true, false );
+        return new ImplementationType( type, true, false, null );
     }
 
-    public static ImplementationType withLoadFactorAdjustment(Type type) {
-        return new ImplementationType( type, true, true );
+    public static ImplementationType withLoadFactorAdjustment(Type type, String factoryMethodName) {
```

Comment:
Instead of using `withLoadFactorAdjustment`, I think we could use an explicit method to deal with factory methods: 

```java
    public static ImplementationType withFactoryMethod(Type type, String factoryMethodName) {
        return new ImplementationType( type, true, true, factoryMethodName );
    }
```

In the end, one concept may or not be related with the other.